spring-security-cas-extension/src/main/java/com/kakawait/spring/security/cas/client/CasStatefulService.java

```diff
@@ -0,0 +1,17 @@
+package com.kakawait.spring.security.cas.client;
+
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.IOException;
+import java.security.Principal;
+
+/**
+ * @author Jonathan Coueraud
+ */
+public interface CasStatefulService {
+
+    HttpRequest createRequest(Principal principal, HttpRequest request) throws IOException;

```

Comment:
I will be more about `EnhanceRequest` or something else rather than `createRequest` because you will not concretely create the request